src/ape/managers/chain.py

```diff
@@ -0,0 +1,360 @@
+import time
+from typing import Dict, Iterator, List, Optional, Union
+
+from dataclassy import dataclass
+
+from ape.api import AddressAPI, BlockAPI, ProviderAPI, ReceiptAPI, TestProviderAPI
+from ape.exceptions import ChainError, ProviderNotConnectedError, UnknownSnapshotError
+from ape.logging import logger
+from ape.types import AddressType, BlockID, SnapshotID
+
+from .networks import NetworkManager
+
+
+@dataclass
+class BlockContainer:
+    """
+    A list of blocks on the chain.
+
+    Usages example::
+
+        from ape import chain
+
+        latest_block = chain.blocks[-1]
+
+    """
+
+    networks: NetworkManager
+    _poll_wait_interval = 2  # Seconds
+
+    def __getitem__(self, block_number: int) -> BlockAPI:
+        """
+        Get a block by number. Negative numbers start at the chain head and
+        move backwards. For example, ``-1`` would be the latest block and
+        ``-2`` would be the block prior to that one, and so on.
+
+        Args:
+            block_number (int): The number of the block to get.
+
+        Returns:
+            :class:`~ape.api.providers.BlockAPI`
+        """
+        if block_number < 0:
+            block_number = self.height + 1 + block_number
+
+        return self._get_block(block_number)
+
+    def __len__(self) -> int:
+        """
+        The number of blocks in the chain.
+
+        Returns:
+            int
+        """
+
+        return self.height + 1
+
+    def __iter__(self) -> Iterator[BlockAPI]:
+        """
+        Iterate over all the current blocks.
+
+        Returns:
+            Iterator[:class:`~ape.api.providers.BlockAPI`]
+        """
+
+        return self.range()
+
+    @property
+    def head(self) -> BlockAPI:
+        """
+        The latest block.
+        """
+
+        return self._get_block("latest")
+
+    @property
+    def height(self) -> int:
+        """
+        The latest block number.
+        """
+
+        return self.head.number
+
+    @property
+    def _provider(self) -> ProviderAPI:
+        if not self.networks.active_provider:
+            raise ProviderNotConnectedError()
+
+        return self.networks.active_provider
+
+    @property
+    def _network_confirmations(self) -> int:
```

Comment:
Should this be a private property?

Reply:
Private like `__network_confirmations` ? I tend to only use those in extreme cases, like `__lock` or `__private_key`.

However, maybe you are suggesting to make this public? In which case, yeah I agree, I just pushed that change. Please let me know if you meant something else

Reply:
I made it `public` ? Let me know if that is not what you meant!